From 53d1f9e92666c77ec13efaeb5f330d05332a8360 Mon Sep 17 00:00:00 2001
From: fujiwarat <takao.fujiwara1@gmail.com>
Date: Fri, 20 Oct 2017 11:34:55 +0900
Subject: [PATCH 26/33] ui/gtk3: Increase timeout of window filter in
 PropertyPanel

The latest Plasma desktop takes more time to generate _NET_WORKAREA
atom and this patch increases the timeout from 1 minute to 5 minutes
so that PropertyPanel locate within the workarea.

Review URL: https://codereview.appspot.com/336840043
---
 ui/gtk3/propertypanel.vala | 13 +++++++++++--
 1 file changed, 11 insertions(+), 2 deletions(-)

diff --git a/ui/gtk3/propertypanel.vala b/ui/gtk3/propertypanel.vala
index f5d9cff7..b94b2b79 100644
--- a/ui/gtk3/propertypanel.vala
+++ b/ui/gtk3/propertypanel.vala
@@ -40,7 +40,9 @@ public class PropertyPanel : Gtk.Box {
     private uint m_auto_hide_timeout = 10000;
     private uint m_auto_hide_timeout_id = 0;
     private bool m_follow_input_cursor_when_always_shown = false;
-    private const uint MONITOR_NET_WORKAREA_TIMEOUT = 60000;
+    // The timeout indicates milliseconds. 1000 msec == 1 sec
+    private const uint MONITOR_NET_WORKAREA_TIMEOUT = 300000;
+    private uint m_remove_filter_id;
 
     public PropertyPanel() {
         /* Chain up base class constructor */
@@ -390,6 +392,11 @@ public class PropertyPanel : Gtk.Box {
             string aname = m_xdisplay.get_atom_name(xevent.xproperty.atom);
             if (aname == "_NET_WORKAREA" && xevent.xproperty.state == 0) {
                 set_default_location();
+                m_root_window.remove_filter(root_window_filter);
+                if (m_remove_filter_id > 0) {
+                    GLib.Source.remove(m_remove_filter_id);
+                    m_remove_filter_id = 0;
+                }
                 return Gdk.FilterReturn.CONTINUE;
             }
         }
@@ -404,7 +411,9 @@ public class PropertyPanel : Gtk.Box {
 
         m_root_window.add_filter(root_window_filter);
 
-        GLib.Timeout.add(MONITOR_NET_WORKAREA_TIMEOUT, () => {
+        m_remove_filter_id = GLib.Timeout.add(MONITOR_NET_WORKAREA_TIMEOUT,
+                                              () => {
+            m_remove_filter_id = 0;
             m_root_window.remove_filter(root_window_filter);
             return false;
         },
-- 
2.15.1

